java/arcs/core/data/proto/policy.proto

```diff
@@ -0,0 +1,100 @@
+syntax = "proto3";
+
+package arcs;
+
+option java_package = "arcs.core.data.proto";
+option java_multiple_files = true;
+
+// Defines a data usage policy.
+message PolicyProto {
+  // Policy name.
+  string name = 1;
+
+  // Targets governed by the policy.
+  repeated PolicyTargetProto targets = 2;
+
+  // Additional config metadata for the policy.
+  repeated PolicyConfigProto configs = 3;
+
+  // Human readable description for the policy.
+  string description = 4;
+
+  enum EgressType {
+    EGRESS_TYPE_UNSPECIFIED = 0;
+    LOGGING = 1;
+    FEDERATED_AGGREGATION = 2;
+  }
+  // Egress type permitted by the policy.
+  EgressType egress_type = 5;
+}
+
+message PolicyTargetProto {
+  // Schema type of the store targeted by the policy.
+  string schema_type = 1;
+
+  // The maximum age of the data that can be used, as a duration (not timestamp)
+  // in milliseconds.
+  int64 max_age_ms = 2;
+
+  // Valid retention options for storing the target data.
+  repeated PolicyRetentionProto retentions = 3;
+
+  // Allowed usages for all fields in this target type.
+  repeated PolicyFieldProto fields = 4;
+}
+
+message PolicyFieldProto {
+  // Field name.
+  string name = 1;
+
+  enum UsageType {
+    USAGE_TYPE_UNSPECIFIED = 0;
+    // Can be used for anything.
+    ANY = 1;
+    // Can be egressed.
+    EGRESS = 2;
+    // Can be used in joins, but not read.
+    JOIN = 3;
+  }
+
+  message AllowedUsage {
+    // The type of usage that is permitted.
+    UsageType usage = 1;
+
+    // An redaction label that must be applied before the data can be used for
+    // this usage type. Optional: an empty string means no redaction is
+    // required.
+    string redaction_label = 2;
```

Comment:
Shouldn't this be a repeated field? Or, is the idea that we will have `usages` values like `(join, redaction)`, `(join, anotherRedaction)`

Reply:
If i were to interpret that, i would think that 
`(join, [redaction]), (join, [otherredaction])` means the field can be joined with either one of the redactions, but if `(join, [redaction, otherredaction])`, then both must be applied before joining.

Reply:
The policy language currently only allows you to express usages in the form `<label>: [<usages>]`, e.g.:

```
someLabel: [egress, join]
anotherLabel: [*]
```

There's no support (yet) for requiring a combination of labels, or an ordered application of labels. If/when we add those, this proto will need to be updated to represent those ideas. For now though, we will just transform those into a series of AllowedUsages:

```
usages {
  usage: EGRESS
  redaction_label: "someLabel"
}
usages {
  usage: JOIN
  redaction_label: "someLabel"
}
usages {
  usage: ANY
  redaction_label: "anotherLabel"
}
```